test/helpers.py

```diff
@@ -167,3 +168,29 @@ def build_alt_svc_frame(self, stream_id, origin, field):
         f.origin = origin
         f.field = field
         return f
+
+
+def sanity_check_settings_header_data(frames,
+                                      expected_flow_controlled_length,
+                                      expect_padded_flag,
+                                      expected_data_frame_pad_length):
+        assert len(frames) == 3
+        settings_frame = frames[0]
+        header_frame = frames[1]
+        data_frame = frames[2]
+        assert isinstance(
+            settings_frame, hyperframe.frame.SettingsFrame
+        )
+        assert isinstance(header_frame, hyperframe.frame.HeadersFrame)
+        assert isinstance(data_frame, hyperframe.frame.DataFrame)
+
+        assert(
+            (data_frame.flow_controlled_length ==
+             expected_flow_controlled_length)
+        )
+        if expect_padded_flag:
+            assert 'PADDED' in data_frame.flags
+        else:
+            assert 'PADDED' not in data_frame.flags
+
+        assert data_frame.pad_length == expected_data_frame_pad_length
```

Comment:
Given that this is only used in one test location, I think it'd be better to make it a helper function in that file rather than place it in a central location.